### PR TITLE
ARC Feedback Updates and other changes

### PR DESCRIPTION
**Pull request by pathakraul**

### CODE REVIEW

**Comment by pathakraul** [comment]
@avpatel

File: src/service-groups.adoc [hunk 1]
```diff
@@ -42,9 +42,11 @@ should be invoked.
 
 This specification defines standard RPMI service groups and RPMI services
 with the provision to add more service groups as required in the future.
-The platform vendors can provide implementation specific RPMI service groups.
-The <<table_service_groups>> below list all standard RPMI service groups
-defined by this specification.
+The RPMI specification also provides experimental service group IDs space
+for development of service group until a standard service group ID is not
```
Comment (lftan) [inline src/service-groups.adoc:46]: should remove "not"?

File: src/service-groups.adoc [hunk 2]
```diff
@@ -42,9 +42,11 @@ should be invoked.
...
+allocated. The platform vendors can provide implementation specific RPMI
+service groups The <<table_service_groups>> below list all standard RPMI
```
Comment (lftan) [inline src/service-groups.adoc:48]: extra "The"

File: src/message-protocol.adoc [hunk 3]
```diff
@@ -166,6 +166,36 @@ values as corresponding fields in the RPMI request message. The `DATALEN`
...
+NOTE: The message token will help the application processors to keep track of
+the origin of the request when it receives  a response. This is  useful when the
+multiple application processors are sharing the same queues. For example, two
+different application processors may send the same type of request message with
+the same SERVICEGROUP_ID and SERVICE_ID. When the response messages for both
+requests are received from the platform microcontroller, the token helps
+distinguish which response belongs to which request.
+
+NOTE: The RPMI specification recommends monotonically increasing token numbers
+and the token number can be initialized from any value without any constarints.
```
Comment (lftan) [inline src/message-protocol.adoc:178]: Typo 'constarints" --> constraints

File: src/message-protocol.adoc [hunk 4]
```diff
@@ -166,6 +166,36 @@ values as corresponding fields in the RPMI request message. The `DATALEN`
...
+platform microcontroller to ring the doorbell after sending the response back.
+If the `flags[3]` bit is `0` in the request message
+header, it means that the application processor is going poll for the
+response message in the queue and the platform microcontroller should not
+ring the doorbell.
+
+NOTE: The flags[3] bit can be used for a particular message or for the entire
+lifecycle of RPMI message communication. For example if the application
+processor and the platform microcontroller are capable for MSIs and the application
+processor has configured MSI details via defined service in Section 4.1.10 then
```
Comment (lftan) [inline src/message-protocol.adoc:191]: Shouldn't hardcoded "Section 4.1.10", create a label/tag at section Section 4.1.10 (BASE_SET_MSI), then reference it with <<BASE_SET_MSI-lable>>

File: src/transport.adoc [hunk 5]
```diff
@@ -38,6 +38,11 @@ the platform microcontroller can avoid implementing the P2A channel.
 The current RPMI specification only defines a shared memory based transport but
 other transport types can be added in the future.
 
+NOTE: The shared memory for RPMI transport and fast-channels allocated
+in DRAM or in on-chip RAM will require memory attributes configuration. These
+memory attributes also called PMA(Physical Memory Attributes) are defined in
```
Comment (lftan) [inline src/transport.adoc:43]: add space before (

File: src/transport.adoc [hunk 6]
```diff
@@ -69,26 +74,49 @@ processors must discover it using standard hardware description mechanisms
...
-=== Fast-channels
-Fast-channels are special shared memory-based channels used in scenarios
-requiring lower latency and faster processing of requests from application
-processors to the platform microcontroller.
-
-The layout and request format of fast-channels are service group specific
-and only a few service groups may support fast-channels. A service group
-that supports fast-channels:
+NOTE: If the platform supports PLIC, the platform need to provide a MMIO
+register to inject an edge-triggered interrupt.
```
Comment (lftan) [inline src/transport.adoc:81]: Not very understand this.
edge-triggered interrupt is not a must for PLIC, right?
and this MMIO register is a hardcoded csr address?

File: src/transport.adoc [hunk 7]
```diff
@@ -69,26 +74,49 @@ processors must discover it using standard hardware description mechanisms
...
 
-NOTE: To avoid the caching side-effects, the platform can configure the
-fast-channel shared memory as non-cacheable or IO memory for both the
-application processors and the platform microcontroller.
+The layout and data format of a fast-channel are RPMI service specific in a
+service group and defined in the respective service group sections. 
+
+The platform must setup the PMA for the shared memory used for the fast-channels.
+
+NOTE: Its possible that the application processor and the platform
```
Comment (lftan) [inline src/transport.adoc:111]: "It is"

File: src/transport.adoc [hunk 8]
```diff
@@ -69,26 +74,49 @@ processors must discover it using standard hardware description mechanisms
...
+microcontroller are not cache-coherent and using the shared memory may lead to
+caching side effects such as data inconsistency between the platform
+microcontroller and the application processor, write propagation delays and
+others issues which may lead to race conditions. To avoid the caching
+side-effects, the platform can configure the memory attribute of the shared
+memory as non-cacheable or IO memory for both the application processor and the
+platform microcontroller and implementation can perform manual cache maintenance
```
Comment (lftan) [inline src/transport.adoc:118]: Split the long sentence. Eg:
Additionally, the implementation can perform manual cache maintenance using cache flush and invalidate operations.

File: src/transport.adoc [hunk 9]
```diff
@@ -97,9 +125,17 @@ device memory. The RPMI shared memory transport does not specify where the
 shared memory resides in a platform, but it must be accessible from both the
 application processors and the platform microcontroller.
 
-NOTE: To avoid the caching side-effects, the platform can configure the shared
+The platform must setup the PMA for the shared memory used for RPMI transport.
+
+NOTE: Its possible that the application processor and the platform
```
Comment (lftan) [inline src/transport.adoc:130]: Repeat the same Note here?

File: src/transport.adoc [hunk 10]
```diff
@@ -97,9 +125,17 @@ device memory. The RPMI shared memory transport does not specify where the
...
+microcontroller are not cache-coherent and using the shared memory may lead to
+caching side effects such as data inconsistency between the platform
+microcontroller and the application processor, write propagation delays and
+others issues which may lead to race conditions. To avoid the caching
+side-effects, the platform can configure the memory attribute of the shared
 memory as non-cacheable or IO memory for both the application processor and the
-platform microcontroller.
+platform microcontroller and implementation can perform manual cache maintenance
```
Comment (lftan) [inline src/transport.adoc:137]: Same comment as above.

File: src/srvgrp-base.adoc [hunk 11]
```diff
@@ -65,15 +65,28 @@ The following table lists the services in the BASE service group:
 |===
 
 ==== RPMI Implementation IDs
+The RPMI specification defines space for standard implementation IDs and for
+experimental implementation IDs. The experimental implementation IDs can be used
+by the implementations until a standard implementation ID is not assigned to it.
```
Comment (lftan) [inline src/srvgrp-base.adoc:70]: Remove 'not'.

"...until a standard implementation ID is assigned to it."

File: src/transport.adoc (hunk 6 above)
Comment (pathakraul) [reply src/transport.adoc:81]: Edge triggered interrupt is preferred in the Linux. The write to the MMIO register will trigger the interrupt enable bit which is configured for wired interrupt. Its not a CSR address.

File: src/transport.adoc (hunk 9 above)
Comment (pathakraul) [reply src/transport.adoc:130]: That two notes are separately provided one for fastchannel and one for the RPMI transport shared memory

**Comment by pathakraul** [comment]
If there are no comments then I will merge this tomorrow and release it to ARC for review

**Comment by avpatel** [comment]
Looks good to me.

**Comment by lftan** [comment]
Looks good, merged.

**Pull request merged**